### PR TITLE
API calling is and data is dynamic as per json

### PR DESCRIPTION
**Pull request by DebasishPati005**

### CODE REVIEW

File: src/Assingment1API.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                    "source : " + getDomainFromEmail((String) emailDetailObject.get("email")) + "\n" +
+                    "inbox : " + getInboxMailsCarbonEmission((long) emailDetailObject.get("inbox")) + " KG\n" +
+                    "sent : " + getSentMailCarbonEmission((long) emailDetailObject.get("sent")) + " KG\n" +
+                    "spam : " + getSpamMailsCarbonEmission((long) emailDetailObject.get("spam")) + " KG\n" +
+                    "}");
+        }
+    }
+
+    static String getDomainFromEmail(String emailId) {
+        String wholeDomain = emailId.split("@")[1];
```
Comment (Intimetec-Khushal-Agarwal) [inline src/Assingment1API.java]: Only Domain can be used in place of wholeDomain.

File: src/Assingment1API.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        return matchedEmailDetails;
+    }
+
+    public static void calculateAndPrintCarbonEmission(JSONObject emailDetailObject) {
+        if (emailDetailObject == null) {
+            System.out.println("Sorry no email record found with your given id");
+        } else {
+            System.out.println("{\n" +
+                    "source : " + getDomainFromEmail((String) emailDetailObject.get("email")) + "\n" +
+                    "inbox : " + getInboxMailsCarbonEmission((long) emailDetailObject.get("inbox")) + " KG\n" +
```
Comment (Intimetec-Khushal-Agarwal) [inline src/Assingment1API.java:84]: If function return type is double then is it required to typecast in long?

File: src/Assingment1API.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            if (responseCode == 200) {
+                BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+                StringBuilder response = new StringBuilder();
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    response.append(line);
+                }
+                reader.close();
+
+                Object obj = jsonParserObj.parse(response.toString());
```
Comment (Intimetec-Khushal-Agarwal) [inline src/Assingment1API.java]: in place of obj you can use object.

File: src/Assingment1API.java [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+    static double getInboxMailsCarbonEmission(long inboxEmails) {
+        return 0.25 * inboxEmails;
+    }
+
+    static double getSentMailCarbonEmission(long inboxEmails) {
+        return 0.8 * inboxEmails;
+    }
+
+    static double getSpamMailsCarbonEmission(long inboxEmails) {
```
Comment (Intimetec-Khushal-Agarwal) [inline src/Assingment1API.java]: you can use long in place of return type double.
As we are getting inbox mail in long and multiplying it only by integer value.

File: src/Assingment1API.java [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+import org.json.simple.parser.ParseException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Scanner;
+
+public class Assingment1API {
```
Comment (MJmitali) [inline src/Assingment1API.java:13]: class name not clear

Comment (MJmitali) [inline src/Assingment1API.java]: class name not in context

File: src/Assingment1API.java (hunk 3 above)
Comment (MJmitali) [inline src/Assingment1API.java]: variable obj is not explanatory

File: src/Assingment1API.java [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                while ((line = reader.readLine()) != null) {
+                    response.append(line);
+                }
+                reader.close();
+
+                Object obj = jsonParserObj.parse(response.toString());
+                // Get elements within the JSON structure which is provided by jsonbin.io
+                // converted into JSON
+                JSONObject jsonObject = (JSONObject) obj;
+                // System.out.println(jsonObject.get("record") + "\n");
```
Comment (Harshita-Soni23) [inline src/Assingment1API.java]: Please remove commented code.

File: src/Assingment1API.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Scanner;
+
+public class Assingment1API {
+
+    public static void main(String[] args) {
```
Comment (Harshita-Soni23) [inline src/Assingment1API.java]: can we separate the main class in another file and  findEmailAddressDetails, calculateAndPrintCarbonEmission in separate one?

File: src/Assingment1API.java [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+            connection.disconnect();
+        } catch (IOException | ParseException e) {
+            e.printStackTrace();
+        }
+        scan.close();
+    }
+
+    public static JSONObject findEmailAddressDetails(JSONArray emailRecordArray, String emailId) {
+        JSONObject matchedEmailDetails = null;
+        for (Object eachEmailRecord : emailRecordArray) {
```
Comment (MJmitali) [inline src/Assingment1API.java]: can use emailRecord and emailRecords , as Array should not used as a suffix

File: src/Assingment1API.java [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+        JSONParser jsonParserObj = new JSONParser();
+        String apiUrl = "https://api.jsonbin.io/v3/qs/653f79a954105e766fc8fd69";
+        Scanner scan = new Scanner(System.in);
+        System.out.print("Enter email address: ");
+        String emailId = scan.nextLine();
+        System.out.print("Wait data is fetching ");
+
+        try {
+            URL url = new URL(apiUrl);
```
Comment (Harshita-Soni23) [inline src/Assingment1API.java]: can we make a function for the code till line no 31.

File: src/Assingment1API.java [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    response.append(line);
+                }
+                reader.close();
+
+                Object obj = jsonParserObj.parse(response.toString());
+                // Get elements within the JSON structure which is provided by jsonbin.io
+                // converted into JSON
+                JSONObject jsonObject = (JSONObject) obj;
```
Comment (Harshita-Soni23) [inline src/Assingment1API.java]: variable name not explanatory. We don't use object in suffix for a variable.

File: src/Assingment1API.java [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+                // System.out.println(jsonObject.get("record") + "\n");
+
+                // Get elements within the JSON structure which is provided by jsonbin.io
+                JSONArray arrayOfEmails = (JSONArray) jsonObject.get("record");
+                JSONObject matchedEmailObject = findEmailAddressDetails(arrayOfEmails, emailId);
+
+                System.out.print("Success data is fetched! 😜");
+                calculateAndPrintCarbonEmission(matchedEmailObject);
```
Comment (MJmitali) [inline src/Assingment1API.java]: A function should do one work at a time . It can divided into calculateCarbonEmission and printCarbonEmission.

**Review by MJmitali — APPROVED** [review]

**Review by Harshita-Soni23 — APPROVED** [review]

**Pull request merged**